### PR TITLE
chore(871): snapshot test package.json

### LINKED ISSUE #871: Snapshot test package.json
# Description
We use `build-package-json.js` script which copies dependencies from package.json, filters out some of them and create a new one which is used for newskit package in NPM.

However, every time we add a new package we also might need to add it to the ignore list ( dependenciesNotToBeIncluded ).

Wound be nice to have a unit test that checks that so we don't push dependencies to npm when they are not needed.

# Tasks/Steps
- add unit test that takes current package json; run filterPackageJson(json) over it, and make snapshot from output. 

# Acceptance Criteria
- when new dependencies is added, the unit tests fails and snapshot need to be updated.

# Supporting Information
*Process/sequence flows, wireframes, third party documentation, tech feasibility, architecture overview/documentation, high level design, data specs etc...*

# Dependencies
*Any known business dependencies or reasons to engage with another team and input into their roadmap/timings*

### PR DESCRIPTION
**Pull request by JohnTParsons**
closes #871

**What**

1. Background - as a reminder to devs to keep dependenciesNotToBeIncluded up-to-date
2. What did you do - add test
3. What does the reviewers should expect - snapshot test to fail if/when package.json changes

**I have done:**
- [x] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by pp-serviceaccount** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs-pr.s3-website-eu-west-1.amazonaws.com/871-test-package
  Storybook: http://ncu-newskit-docs-pr.s3-website-eu-west-1.amazonaws.com/871-test-package/storybook

**Review by mutebg — DISMISSED** [review]

**Review by Vanals — DISMISSED** [review]

File: src/__tests__/__snapshots__/package.test.ts.snap [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+exports[`package.json updated remind developer that dependenciesNotToBeIncluded array may need updating 1`] = `
+Object {
+  "dependencies": Object {
+    "@capsizecss/core": "^3.0.0",
+    "@emotion-icons/emotion-icon": "4.1.0",
+    "@emotion-icons/material": "3.9.0",
+    "@emotion-icons/material-outlined": "3.8.0",
+    "@floating-ui/react-dom-interactions": "^0.6.0",
+    "@newskit-themes/newskit-website": "1.0.2",
+    "@newskit-themes/the-sun": "^0.1.0",
```
Comment (jps) [inline src/__tests__/__snapshots__/package.test.ts.snap:12]: these can be excluded

File: src/__tests__/package.test.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const {newPackageJson} = require('../../scripts/build-package-json-support');
+
+describe('package.json updated', () => {
```
Comment (jps) [inline src/__tests__/package.test.ts:3]: Good idea adding this test in 🏅

**Review by jps — CHANGES_REQUESTED** [review]

File: src/__tests__/__snapshots__/package.test.ts.snap (hunk 1 above)
Comment (JohnTParsons) [reply src/__tests__/__snapshots__/package.test.ts.snap]: Done

**Review by mutebg — APPROVED** [review]

**Review by jps — APPROVED** [review]

**Pull request merged**